curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/65ca2d0625aa3a3201067f70.md

```diff
@@ -0,0 +1,168 @@
+---
+id: 65ca2d0625aa3a3201067f70
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+In the callback function of your `forEach`, use the `el` parameter to access the `counts` object. If the element is not in `counts`, set it to `0`. Otherwise, increment the count for the element. 
+
+Below your `forEach`, log out the final result of `counts`.
+
+Then change your return statement to return `counts` instead of `array`.
+
+To test this, enter the numbers `4, 4, 2, 5` and click `Calculate`. You should see the following in the console:
+
+```js
+{ '2': 1, '4': 2, '5': 1 }
+```
+
+# --hints--
+
+You should change your return statement to return `counts` instead of `array`.
+
+```js
+assert.match(getMode.toString(), /return\s+counts\s*;/);
+```
+
+Your `.forEach()` method should increment the count for each element. Don't forget the initialization scenario. Enter the numbers `4, 4, 2, 5` and click `Calculate` for testing.
```

Comment:
You are only incrementing if the element is already in the `counts` object. otherwise `counts[el]` should be set to `0`

```suggestion
You should check if the current element can be found inside the `counts` object. If it can be found, update the value of `counts[el]` by `1`. Otherwise, assign the number `0` to `counts[el]`. Enter the numbers `4, 4, 2, 5` and click `"Calculate"` for testing.
```